### PR TITLE
Cache the output of log_js_function

### PR DESCRIPTION
**Pull request by mennopruijssers**
Especially with recursion this will improve performance.

given this code as test:
```
var dummy = new Error().stack; // Persuade v8 to compute line numbers

var MAX_DEPTH = 199;

var runRecursive = function runRecursive(depth, cb) {
    if (depth >= MAX_DEPTH) {
        cb();
    } else {
        runRecursive(depth+1, cb);
   }
}

runRecursive(0, function print(){
    while(true) {
        console.log("Hello!");
    }
});
```

Before caching, systemtap exits with a "probe overhead exceeded threshold"-error; after caching succeeds!

### CODE REVIEW

**Comment by Raynos** [comment]
lgtm.

File: lib/profile_node.stp [hunk 1]
```diff
@@ -200,24 +200,29 @@ function get_shared_info_line_number(shared_info:long) {
...
  * Works for most unicode chars!
  */
-function print_user_utf16_string(data_ptr:long, length:long) {
+function get_user_utf16_string(data_ptr:long, length:long) {
     data_ptr = strip_low_bit(data_ptr)
+    str = ""
     for (i = 0; i < 200 && i < length; i++) {
         c = user_uint16(data_ptr)
-        printf("%c", c);
+        str .= sprintf("%c", c);
```
Comment (CorgiMan) [inline lib/profile_node.stp]: Not sure if it is a big deal and I'm not sure if I'm correct. But for better performance we might want to use a string builder type of thing here

Comment (mennopruijssers) [inline lib/profile_node.stp]: That's not possible in a SystemTap script...

**Comment by danielheller** [comment]
Looking great to me in general, just a couple of comments earlier.  Could you please also squash the cache key improvement with the cache change, so it all lands in one commit?  Thanks a lot!

File: lib/profile_node.stp (hunk 1 above)
Comment (CorgiMan) [inline lib/profile_node.stp]: You can't put the strings in an array and concat the array in one go? It's not a big deal anyway. :)

Comment (mennopruijssers) [inline lib/profile_node.stp]: Not in system tap :(...

**Comment by mennopruijssers** [comment]
I squashed them and reduced the nesting. Do you want to take another look or is it ok to merge it?

**Comment by Raynos** [comment]
lgtm  :)

**Pull request merged**